bin/plugin/commands/performance.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+
+/**
+ * Internal dependencies
+ */
+const { formats, log } = require( '../lib/logger' );
+const {
+	runShellScript,
+	readJSONFile,
+	askForConfirmation,
+	getRandomTemporaryPath,
+} = require( '../lib/utils' );
+const git = require( '../lib/git' );
+const config = require( '../config' );
+
+/**
+ * @typedef WPRawPerformanceResults
+ *
+ * @property {number[]} load             Load Time.
+ * @property {number[]} domcontentloaded DOM Contentloaded time.
+ * @property {number[]} type             Average type time.
+ * @property {number[]} focus            Average block selection time.
+ */
+
+/**
+ * @typedef WPPerformanceResults
+ *
+ * @property {number} load             Load Time.
+ * @property {number} domcontentloaded DOM Contentloaded time.
+ * @property {number} type             Average type time.
+ * @property {number} minType          Minium type time.
+ * @property {number} maxType          Maximum type time.
+ * @property {number} focus            Average block selection time.
+ * @property {number} minFocus         Min block selection time.
+ * @property {number} maxFocus         Max block selection time.
+ */
+/**
+ * @typedef WPFormattedPerformanceResults
+ *
+ * @property {string} load             Load Time.
+ * @property {string} domcontentloaded DOM Contentloaded time.
+ * @property {string} type             Average type time.
+ * @property {string} minType          Minium type time.
+ * @property {string} maxType          Maximum type time.
+ * @property {string} focus            Average block selection time.
+ * @property {string} minFocus         Min block selection time.
+ * @property {string} maxFocus         Max block selection time.
+ */
+
+/**
+ * Computes the average number from an array numbers.
+ *
+ * @param {number[]} array
+ *
+ * @return {number} Average.
+ */
+function average( array ) {
+	return array.reduce( ( a, b ) => a + b ) / array.length;
+}
+
+/**
+ * Computes the median number from an array numbers.
+ *
+ * @param {number[]} array
+ *
+ * @return {number} Median.
+ */
+function median( array ) {
+	const mid = Math.floor( array.length / 2 ),
+		numbers = [ ...array ].sort( ( a, b ) => a - b );
+	return array.length % 2 !== 0
+		? numbers[ mid ]
+		: ( numbers[ mid - 1 ] + numbers[ mid ] ) / 2;
+}
+
+/**
+ * Rounds and format a time passed in milliseconds.
+ *
+ * @param {number} number
+ *
+ * @return {string} Formatted time.
+ */
+function formatTime( number ) {
+	const factor = Math.pow( 10, 2 );
```

Comment:
I'm guessing there's a good reason we're calling `Math.pow( 10, 2 )` instead of simply using `100` but I can't figure out what it might be. Any hints @youknowriad?

Reply:
also I see this is called `formatTime` but it appears to be applied to a number of different data types and it appears to be used as a way of converting to a fixed number of decimal places.

in `trunk` the `' ms'` suffix is gone, moved to where we store this data in a JSON file. I think it's currently the equivalent of this, but I'm confused because of the disparity between what it's called and what it does.

```js
mapValues( medians, n => Math.round( 100 * n ) / 100 )
```

Though to be honest if this _is_ all we're doing then would you find it wrong to leave the numbers as they are and then run `.toFixed(2)` at the place where we print them? I think that could make the intention more clear (if our goal is simply to limit to a fixed number of decimal places when we print these out) and leave the data more raw along the way.

Reply:
>  the intention more clear (if our goal is simply to limit to a fixed number of decimal places when we print these out) and leave the data more raw along the way.

I think that's probably the intent here yeah.